db/migrate/20200726220247_migrate_distributor_to_new_distributor.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class MigrateDistributorToNewDistributor < ActiveRecord::Migration[6.0]
+  def change
+    add_reference :distributors, :contact, index: true
+    add_reference :campaigns, :distributor, index: true
+
+    sellers = Seller.all
+    sellers.each do |seller|
+      contact = seller.distributor
+
+      # If there is a distributor for the Seller, then there was a previous
+      # campaign. So far, only 46 Mott and Melonpanna should have had previous
+      # campaigns
+      next unless contact.present?
+
+      # NB(justintmckibben): Remember to update the end_date of past campaigns
+      # to the real end_date
+      campaign = Campaign.create valid: true, end_date: Time.now
+      campaign.seller = seller
+      campaign.distributor = Distributor.create contact: contact
+    end
+
+    remove_column :contacts, :seller_id
```

Comment:
Maybe can push this off to a future migration depending on how deploys are done. I'll follow up in Slack.